pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -96,7 +96,7 @@ func (ds *dockerService) RunPodSandbox(ctx context.Context, r *runtimeapi.RunPod
 	}
 
 	// Step 2: Create the sandbox container.
-	if r.GetRuntimeHandler() != "" {
+	if r.GetRuntimeHandler() != "" && r.GetRuntimeHandler() != runtimeName {
```

Comment:
```suggestion
                runtimeHandlerEmpty := r.GetRuntimeHandler() == ""
                runtimeHandlerIsDocker := r.GetRuntimeHandler() == runtimeName
		if !runtimeHandlerEmpty && !runtimeHandlerIsDocker {
```

small nit, but could this be slightly more readable with a descriptive variable name?